### ISSUE TITLE
Dereference error 

### ISSUE THREAD

**Issue by fbrausse** [esbmc/esbmc#575]
This code
```c
#include <assert.h>
#include <stdlib.h>

typedef struct str {
    unsigned field1 : 15;
} str;

int main(void) {

  str array, *ptr = &array;
  int i = 0;
  ptr[i].field1 = i;
  unsigned n = ptr->field1;
  assert(n == i);

  return 0;
}
```
 is wrongly attributed a
```
Counterexample:

State 1 file fedor-bitfields-01_ptr_bf_1.c line 12 function main thread 0
----------------------------------------------------
  (ptr + (signed long int)i)->field1 = byte_update_little_endian(array.field1, 0, 0 | byte_extract_little_endian(array.field1, 0) & 128)

State 2 file fedor-bitfields-01_ptr_bf_1.c line 13 function main thread 0
----------------------------------------------------
  n = (unsigned int)array.field1

State 3 file fedor-bitfields-01_ptr_bf_1.c line 14 function main thread 0
----------------------------------------------------
Violated property:
  file fedor-bitfields-01_ptr_bf_1.c line 14 function main
  assertion n == i
  n == (unsigned int)i


VERIFICATION FAILED
```
Replacing `[i].` with either `[0].` or `->` fixes the issue, as does making the bitfield width a multiple of 8.

**Comment by lucasccordeiro** [comment]
@shmarovfedor: Can you please take a look at this issue? It seems to be related to the bitfield support.

**Comment by shmarovfedor** [comment]
Yea it looks like the case of bitfields dereferenced using a dynamic offset (introduced by `[i]`) is not handled correctly in the `dereference` module. I will be working on it.

**Comment by fbrausse** [comment]
Fixed by #624.
